hack/release.sh

```diff
@@ -0,0 +1,107 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -euo pipefail
+
+# The script will try to checkout the release version if specified.
+# This is useful for manual or older tagged version releases.
+# If a release version is not specified, it will check the latest commit
+# for a tag, if a tag has appeared on the latest commit, it will build
+# and push the binaries from the latest commit.
+RELEASE_VERSION=${1:-}
+RELEASE_TAG=${RELEASE_VERSION:-}
+CHECKOUT_OLDER="${2:-}"
+GCP_PROJECT=${GCP_PROJECT:-k8s-jkns-e2e-gce-federation}
+GCS_BUCKET=${GCS_BUCKET:-kubernetes-federation-release}
+GCR_REPO_PATH="gcr.io/google_containers/fcp-amd64"
+SCRIPT_ROOT="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+TMPDIR="$(mktemp -d /tmp/k8s-fed-relXXXXXXXX)"
+KUBE_ROOT=${SCRIPT_ROOT}/..
+
+source "${KUBE_ROOT}/hack/lib/version.sh"
+
+function clean_up() {
+  if [[ "${TMPDIR}" == "/tmp/k8s-fed-rel"* ]]; then
+    rm -rf "${TMPDIR}"
+  fi
+}
+trap clean_up EXIT
+
+cd "${SCRIPT_ROOT}/.."
+
+if [[ -z ${RELEASE_VERSION} ]]; then
+  echo "No release tag specified, will check if the latest commit has a tag."
+  tags=$(git describe --exact-match)
+  # Just use the first one in the list if the commit has multiple tags.
+  RELEASE_TAG=${tags[0]}
+else
+  git checkout ${RELEASE_VERSION}
+fi
+
+# Load the version vars.
+kube::version::get_version_vars
+
+# This is to prevent the script from starting if its on a dirty commit.
+if [[ "${KUBE_GIT_VERSION}" != "${RELEASE_TAG}" ]]; then
+  echo "Version being build: ${KUBE_GIT_VERSION} does not match the release version: ${RELEASE_TAG}, there probably is uncommited work."
+  exit 1
+fi
+
+# Check for and install (some) necessary dependencies.
+# TODO: irfanurrehman we are yet to move completely to bazel, which can
+# avoid docker as dependency.
+command -v docker >/dev/null 2>&1 || { echo >&2 "Please install docker before running this script."; exit 1; }
+command -v gcloud >/dev/null 2>&1 || { echo >&2 "Please install gcloud before running this script."; exit 1; }
+gcloud components install gsutil
+
+RELEASE_TMPDIR="${TMPDIR}/${RELEASE_TAG}"
+
+# Build the tarballs of the tools.
+make quick-release
+
+# Copy the archives.
+mkdir -p "${RELEASE_TMPDIR}"
+cp \
+  _output/release-tars/federation-client-linux-amd64.tar.gz \
+  _output/release-tars/federation-server-linux-amd64.tar.gz \
+  "${RELEASE_TMPDIR}"
+
+# Create the `latest` file.
+echo "${RELEASE_VERSION}" > "${TMPDIR}/latest"
+
+pushd "${RELEASE_TMPDIR}" 1>&2
+
+# extract tar, load fcp image and push to gcr.
+tar -xf federation-server-linux-amd64.tar.gz 1>&2
+gcloud docker -- load -i federation/fcp-amd64.tar
+gcloud docker -- push "${GCR_REPO_PATH}:${RELEASE_TAG}"
```

Comment:
`RELEASE_TAG` could be empty is some flows. I think docker push may fail in such cases.

Reply:
It exits as in previous comment, if release tag is empty!